modules/wallet/transactionbuilder.go

```diff
@@ -634,3 +634,127 @@ func (w *Wallet) RegisterTransaction(t types.Transaction, parents []types.Transa
 func (w *Wallet) StartTransaction() modules.TransactionBuilder {
 	return w.RegisterTransaction(types.Transaction{}, nil)
 }
+
+// SpendableOutputs returns the outputs spendable by the wallet.
+func (w *Wallet) SpendableOutputs() []modules.SpendableOutput {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	// ensure durability of reported outputs
+	w.syncDB()
```

Comment:
Do we need to sync here? Once you got the outputs and created a transaction, the transaction pool will decide if the transaction is valid. Therefore I think it is not really an issue if the output disappears from the wallet due to a crash.

Reply:
it's more about the principle. If the wallet reports that it has an output, and then it crashes, it should still have that output next time you ask it. The general rule is: "don't give the user any data that won't survive a crash."

Reply:
I see :+1: